feat: display ChatGPT model options

samantha-t28 opened the pull request:
### Feature: Display ChatGPT Models
- Changed label from "Model" to "Select Model" to improve clarity for users when choosing a model.
- Incorporated the datalist element to display model options.
- Added instructions for available model options and a link to the full list of models

### Issue with Default Value
<img width="681" alt="Screenshot 2024-10-26 at 11 30 33 PM" src="https://github.com/user-attachments/assets/d5303419-62f5-4398-8f7a-775ea1411640">

* When a default AI model is set, the <datalist> dropdown only displays the current model.
* The full list of options isn't shown unless the user clears the input, potentially causing confusion.
* <datalist> behaves in a way that if the input matches one of the options exactly, the dropdown might not show the full list because it assumes the selection is already complete.

### Solution:
<img width="691" alt="Screenshot 2024-10-26 at 11 27 24 PM" src="https://github.com/user-attachments/assets/5fe225f4-e16e-4b54-8a89-b8c8ca1ab0e8">

* Removed the default OpenAI model.

<img width="682" alt="Screenshot 2024-10-26 at 11 39 26 PM" src="https://github.com/user-attachments/assets/aed9e2ac-c76d-4741-ab09-acf5a0da51f7">

* The <datalist> dropdown displays all options when the input is clicked, making it easier for the user to browse and select a model.
* By removing the default AI model from the input improves the user experience by making it clearer that the user needs to select a model from the drop down list.

it closes #5

samantha-t28 committed:
feat: display ChatGPT model options

- Changed label from "Model" to "Select Model" to improve clarity for users when choosing a model.
- Incorporated the datalist element to display model options.
- Added instructions for available model options and a link to the full list of models

opauloh commented on src/App.tsx:
```diff
@@ -138,15 +138,31 @@ const SettingsComponent: React.FC = () => {
               <br />
               <input
                 type="text"
-                value={defaultOpenAiModel}
```
if we remove the value, the app won't display it when closing and opening the browser. Can you introduce it back?

The behavior, of not showing all the options once one option is typed is acceptable on inputs with auto-completion and will be already a huge improvement from the previous version.

samantha-t28 replied:
Re-added the default value now in e74b7fbf2bbb9ee2a438968d4aaf99fed6bb64a1

opauloh commented on src/App.tsx:
```diff
@@ -138,15 +138,31 @@ const SettingsComponent: React.FC = () => {
+            <b>gpt-4 </b>, and <b>gpt-3.5-turbo</b>
+            </p>
+          
+           <p>For a complete list of models, please check the <a href="https://platform.openai.com/docs/models" target="_blank">ChatGPT models page</a>.</p>
```
Can we store the URL into a constant? it will make a lot easier to update in the future in case it changes!

samantha-t28 replied:
Stored it now into constants.ts in d8fbe22b10e7c279a6f55a5f4ef7d1120930f1bb

opauloh commented on src/App.tsx:
```diff
@@ -138,15 +138,31 @@ const SettingsComponent: React.FC = () => {
+            <b>gpt-4 </b>, and <b>gpt-3.5-turbo</b>
+            </p>
+          
+           <p>For a complete list of models, please check the <a href="https://platform.openai.com/docs/models" target="_blank">ChatGPT models page</a>.</p>
```
thank you for adding `target="_blank"` attribute, can you also add `rel="noopener noreferrer"` for enhanced security?

Context:

The `rel=noopener` attribute instructs the browser to navigate to the target resource without granting the new browsing context access to the document that opened it.

The `rel=noreelements` instructs the browser, when navigating to the target resource, to omit the [Referer](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer) header and otherwise leak no referrer information. This way the user privacy is not leaked.

Including both ensures security and privacy when navigating to external pages.

samantha-t28 replied:
Added rel="noopener noreferrer" now in a2bc78ef879d93c805c60e5d1cb33b901152d85c 
made a note now about security implications of not having this here!

opauloh reviewed (changes requested):
The code overall looks good, I had two improvement suggestions, and one change requested.

Thank you a lot @samantha-t28, for your work on this first contribution, please feel free to re-request the review once the requested changes are pushed!

samantha-t28 committed:
Adding export const CHATGPT_MODELS_URL

samantha-t28 committed:
Re-added default value to input area

samantha-t28 committed:
Adds security refs rel="noopener noreferrer"

samantha-t28 commented:
d8fbe22b10e7c279a6f55a5f4ef7d1120930f1bb

opauloh commented:
Thank you @samantha-t28 for addressing the comments. 



> ## [![Quality Gate Passed](https://camo.githubusercontent.com/f7c890f34e0efe8252e17e9afd5a85aa22fc9c81b42a0ec3d4821af9f55f250d/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636865636b732f5175616c6974794761746542616467652f71672d7061737365642d323070782e706e67)](https://sonarcloud.io/dashboard?id=luniai_github-code-review&pullRequest=10) **Quality Gate passed**
> Issues ![](https://camo.githubusercontent.com/187ba88c264795e888bdd18141ef29390601e5110d4e9c8cc48f2f8afc183ba0/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f7061737365642d313670782e706e67) [3 New issues](https://sonarcloud.io/project/issues?id=luniai_github-code-review&pullRequest=10&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true) ![](https://camo.githubusercontent.com/86e5c40907bf9e8933fe6bcf185f7076622a76a3bef4b6d2b5fd583959409aef/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f61636365707465642d313670782e706e67) [0 Accepted issues](https://sonarcloud.io/project/issues?id=luniai_github-code-review&pullRequest=10&issueStatuses=ACCEPTED)
> 
> Measures ![](https://camo.githubusercontent.com/187ba88c264795e888bdd18141ef29390601e5110d4e9c8cc48f2f8afc183ba0/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f7061737365642d313670782e706e67) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=luniai_github-code-review&pullRequest=10&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true) ![](https://camo.githubusercontent.com/187ba88c264795e888bdd18141ef29390601e5110d4e9c8cc48f2f8afc183ba0/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f7061737365642d313670782e706e67) [0.0% Coverage on New Code](https://sonarcloud.io/component_measures?id=luniai_github-code-review&pullRequest=10&metric=new_coverage&view=list) ![](https://camo.githubusercontent.com/187ba88c264795e888bdd18141ef29390601e5110d4e9c8cc48f2f8afc183ba0/68747470733a2f2f736f6e6172736f757263652e6769746875622e696f2f736f6e6172636c6f75642d6769746875622d7374617469632d7265736f75726365732f76322f636f6d6d6f6e2f7061737365642d313670782e706e67) [0.0% Duplication on New Code](https://sonarcloud.io/component_measures?id=luniai_github-code-review&pullRequest=10&metric=new_duplicated_lines_density&view=list)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=luniai_github-code-review&pullRequest=10)

I see it passed on the Quality Gates configured in the CI, however 3 small issues where introduced, would you mind addressing them? You should be able to click on the [link](https://sonarcloud.io/project/issues?id=luniai_github-code-review&pullRequest=10&issueStatuses=OPEN,CONFIRMED&sinceLeakPeriod=true) provided to visualize the issues and instructions on how to fix it.

samantha-t28 committed:
Resolved issues flagged by SonarCloud

- Moved available model names to Constant.ts for cleaner code organization.
- Used map to loop through models for improved readability.
- Adjusted formatting to resolve SonarCloud's "ambiguous spacing" warning

opauloh reviewed (approved):
I see you addressed all the comments and Quality Gates issues. Thank you a lot for your contribution to this project 🚀
